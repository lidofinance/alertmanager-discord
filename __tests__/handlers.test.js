const axios = require("axios");
jest.mock("axios");

const { handleHook } = require("../handlers");

function getAlert(i) {
  return {
    status: "resolved",
    labels: { alertname: "activate" },
    annotations: {
      description: "cinema",
      summary: "donor" + String(i).padStart(2, "0"),
    },
  };
}

test("hook works", async () => {
  // mock
  const ctx = {
    routes: { test: "/dev/null" },
    params: { slug: "test" },

    request: {
      body: {
        alerts: Array.from({ length: 11 }, (_, i) => getAlert(i)),
      },
    },
  };

  axios.post.mockResolvedValue(null);

  await handleHook(ctx, () => {});

  expect(ctx.status).toBe(200);
  expect(axios.post.mock.calls.length).toBe(2);
  expect(axios.post.mock.calls).toMatchSnapshot();
});
