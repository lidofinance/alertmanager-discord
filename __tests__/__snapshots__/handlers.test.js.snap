// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`healthcheck works 1`] = `
Array [
  Array [
    "/dev/null",
  ],
]
`;

exports[`hook works (mentions) 1`] = `
Array [
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user0> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor00",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user1> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor01",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user2> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor02",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user3> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor03",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user4> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor04",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user5> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor05",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user6> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor06",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user7> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor07",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user8> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor08",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user9> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor09",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "allowed_mentions": Object {
        "parse": Array [
          "users",
        ],
      },
      "content": "<@user10> <@42>",
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor10",
        },
      ],
    },
  ],
]
`;

exports[`hook works (missing annotations subfield) 1`] = `
Array [
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": undefined,
          "title": "here was no description",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "here was no summary",
          "title": undefined,
        },
      ],
    },
  ],
]
`;

exports[`hook works (no mentions) 1`] = `
Array [
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor00",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor01",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor02",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor03",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor04",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor05",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor06",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor07",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor08",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor09",
        },
      ],
    },
  ],
  Array [
    "/dev/null",
    Object {
      "embeds": Array [
        Object {
          "color": 51283,
          "description": "cinema",
          "title": "donor10",
        },
      ],
    },
  ],
]
`;
